x-pack/plugin/sql/sql-proto/src/main/java/org/elasticsearch/xpack/sql/proto/Payloads.java

```diff
@@ -0,0 +1,253 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.sql.proto;
+
+import com.fasterxml.jackson.core.JsonGenerator;
+import com.fasterxml.jackson.core.JsonParser;
+
+import org.elasticsearch.xpack.sql.proto.content.ConstructingObjectParser;
+import org.elasticsearch.xpack.sql.proto.content.GeneratorUtils;
+import org.elasticsearch.xpack.sql.proto.content.ParseException;
+import org.elasticsearch.xpack.sql.proto.content.ParserUtils;
+import org.elasticsearch.xpack.sql.proto.core.TimeValue;
+
+import java.io.IOException;
+import java.time.ZoneId;
+import java.util.List;
+import java.util.function.Function;
+
+import static org.elasticsearch.xpack.sql.proto.CoreProtocol.BINARY_FORMAT_NAME;
+import static org.elasticsearch.xpack.sql.proto.CoreProtocol.CATALOG_NAME;
+import static org.elasticsearch.xpack.sql.proto.CoreProtocol.CLIENT_ID_NAME;
+import static org.elasticsearch.xpack.sql.proto.CoreProtocol.COLUMNAR_NAME;
+import static org.elasticsearch.xpack.sql.proto.CoreProtocol.COLUMNS_NAME;
+import static org.elasticsearch.xpack.sql.proto.CoreProtocol.CURSOR_NAME;
+import static org.elasticsearch.xpack.sql.proto.CoreProtocol.FETCH_SIZE_NAME;
+import static org.elasticsearch.xpack.sql.proto.CoreProtocol.FIELD_MULTI_VALUE_LENIENCY_NAME;
+import static org.elasticsearch.xpack.sql.proto.CoreProtocol.ID_NAME;
+import static org.elasticsearch.xpack.sql.proto.CoreProtocol.INDEX_INCLUDE_FROZEN_NAME;
+import static org.elasticsearch.xpack.sql.proto.CoreProtocol.IS_PARTIAL_NAME;
+import static org.elasticsearch.xpack.sql.proto.CoreProtocol.IS_RUNNING_NAME;
+import static org.elasticsearch.xpack.sql.proto.CoreProtocol.KEEP_ALIVE_NAME;
+import static org.elasticsearch.xpack.sql.proto.CoreProtocol.KEEP_ON_COMPLETION_NAME;
+import static org.elasticsearch.xpack.sql.proto.CoreProtocol.MODE_NAME;
+import static org.elasticsearch.xpack.sql.proto.CoreProtocol.PAGE_TIMEOUT_NAME;
+import static org.elasticsearch.xpack.sql.proto.CoreProtocol.PARAMS_NAME;
+import static org.elasticsearch.xpack.sql.proto.CoreProtocol.QUERY_NAME;
+import static org.elasticsearch.xpack.sql.proto.CoreProtocol.REQUEST_TIMEOUT_NAME;
+import static org.elasticsearch.xpack.sql.proto.CoreProtocol.ROWS_NAME;
+import static org.elasticsearch.xpack.sql.proto.CoreProtocol.TIME_ZONE_NAME;
+import static org.elasticsearch.xpack.sql.proto.CoreProtocol.WAIT_FOR_COMPLETION_TIMEOUT_NAME;
+import static org.elasticsearch.xpack.sql.proto.content.AbstractObjectParser.ValueType.OBJECT_ARRAY;
+import static org.elasticsearch.xpack.sql.proto.content.ConstructingObjectParser.constructorArg;
+import static org.elasticsearch.xpack.sql.proto.content.ConstructingObjectParser.optionalConstructorArg;
+
+public final class Payloads {
```

Comment:
This class handles parsing and generation for sql-proto. Previously this was nested inside each class which was good for visibility but had issue in reusing the code from different places since it required instance access.